mineral-ui/mineral-ui
# [Flex,FlexItem] Flex gutter refactor & improvements; FlexItem can be a Flex

### Description

- Refactor Flex's gutters
    - To better respect RTL and responsive properties
    - To remove the outer negative margins, which had caused layout issues for consumers and was a prerequisite for...
- Add a `flex` prop to FlexItem to compose Flex instead of Box, which makes it much easier to nest flex containers and flexed items for layout purposes
- Simplify the demo-only styles for Flex, which fixed #710 
- Add some specs for critical/complex functionality
- Update snapshots
    - Pagination snapshots were affected because it uses Flex components
    - Table snapshots were affected because the controlled examples use Flex components

### Motivation and context

Closes #706, closes #710, closes #711

### Screenshots, videos, or demo, if appropriate


https://flex-improvements--mineral-ui.netlify.com/components/flex

### How to test


1. `cd mineral-ui && npm start`
2. Navigate to [Flex](https://flex-improvements--mineral-ui.netlify.com/components/flex
) and [FlexItem](https://flex-improvements--mineral-ui.netlify.com/components/flex-item)
3. Confirm everything looks correct, specifically noting the new [flex props example](https://flex-improvements--mineral-ui.netlify.com/components/flex-item/flex-props)

### Types of changes

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

Breaking: The removal of the negative margins on Flex could have impact on consumer's layout, so I marked it as a breaking change.

### Checklist




### How does this PR make you feel?

![Flex](https://media.giphy.com/media/PFNN9SAJVRYZ2/giphy.gif)
